lib/charms/kubernetes_charm_libraries/v0/multus.py

```diff
@@ -150,8 +150,16 @@ class NetworkAnnotation:
 
     name: str
     interface: str
+    mac: Optional[str] = None
+    ips: Optional[List[str]] = None
 
-    dict = asdict
+    def dict(self) -> dict:
+        """Returns a NetworkAnnotation in the form of a dictionary.
+
+        Returns:
+            dict: Dictionary representation of the NetworkAnnotation
+        """
+        return {key: value for key, value in asdict(self).items() if value}
```

Comment:
Why this change?

Reply:
That's because we don't want `None`s in the dict. If we had `None`s, then the StatefulSet annotation could look something like:
```
"k8s.v1.cni.cncf.io/networks": {"name": "what", "interface": "ever", "mac": null, "ips": null}
```
I figured it'd be better to filter out `None`s and keep the annotation clean.